compiler/rustc_middle/src/ty/query/plumbing.rs

```diff
@@ -259,52 +256,54 @@ macro_rules! define_queries {
 
         #[allow(nonstandard_style)]
         #[derive(Clone, Debug)]
-        pub enum Query<$tcx> {
-            $($(#[$attr])* $name($($K)*)),*
+        pub struct Query {
+            pub name: &'static str,
+            hash: Fingerprint,
+            description: Cow<'static, str>,
+            span: Option<Span>,
         }
 
-        impl<$tcx> Query<$tcx> {
-            pub fn name(&self) -> &'static str {
-                match *self {
-                    $(Query::$name(_) => stringify!($name),)*
-                }
-            }
-
-            pub fn describe(&self, tcx: TyCtxt<$tcx>) -> Cow<'static, str> {
-                let (r, name) = match *self {
-                    $(Query::$name(key) => {
-                        (queries::$name::describe(tcx, key), stringify!($name))
-                    })*
+        impl Query {
+            $(#[allow(nonstandard_style)] $(#[$attr])*
```

Comment:
Can we add a comment here that's something like "generate constructors for Query structs for each query"? It took me a bit to understand what was going on here.

Reply:
Done.